crypto/threads_pthread.c

```diff
@@ -655,6 +842,20 @@ int CRYPTO_THREAD_unlock(CRYPTO_RWLOCK *lock)
 # ifdef USE_RWLOCK
     if (pthread_rwlock_unlock(lock) != 0)
         return 0;
+#  ifdef REPORT_RWLOCK_CONTENTION
+    {
+        struct stack_traces *traces = CRYPTO_THREAD_get_local(&thread_contention_data);
+
+        if (contention_fp == NULL)
```

Comment:
would it be possible to invert the condition here to `contention_fp != NULL` here?

the thing is this approach makes code more fragile with respect to future changes. consider someone adds a code to common location at the end of the function just before the `return 1;`